### PR TITLE
Fix/metrics performance query

### PR DESCRIPTION
**Pull request by gsps1**
JIRA : https://issues.cask.co/browse/CDAP-6860
Build : http://builds.cask.co/browse/CDAP-DUT4581-2

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java [hunk 1]
```diff
@@ -146,16 +150,52 @@ public void add(List<Fact> facts) {
...
+      long secondId = measureNameToEntityIdMap.get(second);
+      if (firstId < secondId) {
+        return -1;
+      } else if (firstId == secondId) {
+        return 0;
+      } else {
+        return 1;
+      }
+    }
+  }
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java:164]: newline between methods

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java [hunk 2]
```diff
@@ -146,16 +150,52 @@ public void add(List<Fact> facts) {
     }
   }
 
+  class MeasureNameComparator implements Comparator<String> {
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: can this be private?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java [hunk 3]
```diff
@@ -146,16 +150,52 @@ public void add(List<Fact> facts) {
...
+    private final Map<String, Long> measureNameToEntityIdMap;
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java:154]: newline between variables and methods

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java [hunk 4]
```diff
@@ -146,16 +150,52 @@ public void add(List<Fact> facts) {
...
+    private MeasureNameComparator(Map<String, Long> measureNameToEntityIdMap) {
+      this.measureNameToEntityIdMap = measureNameToEntityIdMap;
+    }
+
+    @Override
+    public int compare(String first, String second) {
+      long firstId = measureNameToEntityIdMap.get(first);
+      long secondId = measureNameToEntityIdMap.get(second);
+      if (firstId < secondId) {
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: can just do Long.compare(firstId, secondId)

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java [hunk 5]
```diff
@@ -146,16 +150,52 @@ public void add(List<Fact> facts) {
...
 
+  class MeasureNameComparator implements Comparator<String> {
+    private final Map<String, Long> measureNameToEntityIdMap;
+    private MeasureNameComparator(Map<String, Long> measureNameToEntityIdMap) {
+      this.measureNameToEntityIdMap = measureNameToEntityIdMap;
+    }
+
+    @Override
+    public int compare(String first, String second) {
+      long firstId = measureNameToEntityIdMap.get(first);
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: these are guaranteed to be non-null?

Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: ok I see, they are. doesn't this generate warnings though?

**Comment by albertshau** [comment]
minor comments, otherwise lgtm

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java (hunk 2 above)
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: yeah, made it private. used only in this class.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java (hunk 5 above)
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java]: i don't see any warnings in the IDE.

**Comment by albertshau** [comment]
lgtm

**Comment by gsps1** [comment]
build green, merging.

**Pull request merged**